src/v/rpc/connection_cache.cc

```diff
@@ -61,11 +61,14 @@ ss::future<> connection_cache::remove(model::node_id n) {
 /// \brief closes all client connections
 ss::future<> connection_cache::stop() {
     auto units = co_await _mutex.get_units();
-    co_await parallel_for_each(_cache, [](auto& it) {
+    // Exchange ensures the cache is invalidated and concurrent
+    // accesses wait on the mutex to populate new entries.
```

Comment:
we were holding the lock before so wouldn't the concurrent accesses have been waiting (or maybe the cache was checked without acquiring the lock)?